AEPCore/Sources/eventhub/Event.swift

```diff
@@ -39,28 +39,38 @@ public class Event: NSObject, Codable {
 
     /// Event description used for logging
     @objc override public var description: String {
-        // swiftformat:disable indent
-        return "\n[\n" +
-                "  id: \(id.uuidString)\n" +
-                "  name: \(name)\n" +
-                "  type: \(type)\n" +
-                "  source: \(source)\n" +
-                "  data: \(PrettyDictionary.prettify(data))\n" +
-                "  timestamp: \(timestamp.description)\n" +
-                "  responseId: \(String(describing: responseID?.uuidString))\n" +
-                "]"
-        // swiftformat:enable indent
+        return """
+            [
+              id: \(id.uuidString)
+              name: \(name)
+              type: \(type)
+              source: \(source)
+              data: \(PrettyDictionary.prettify(data))
+              timestamp: \(timestamp.description)
+              responseId: \(String(describing: responseID?.uuidString))
+              mask: \(String(describing: mask))
+            ]
+        """
     }
 
+    /// Specifies the properties in the Event and its `data` that should be used in the hash for `EventHistory` storage.
+    @objc public private(set) var mask: [String]?
+
+    /// A calculated hash that represents this Event as defined by its properties and the provided `mask`
+    @objc public lazy var eventHash: UInt32 = {
+        return data?.fnv1a32(mask: mask) ?? 0
+    }()
+
     /// Creates a new `Event` with the given parameters
     /// - Parameters:
     ///   - name: Name for the `Event`
     ///   - type: `EventType` for the `Event`
     ///   - source: `EventSource` for the `Event`
     ///   - data: Any associated data with this `Event`
+    ///   - mask: Defines which properties should be used in creation of the Event's hash
     @objc
-    public convenience init(name: String, type: String, source: String, data: [String: Any]?) {
-        self.init(name: name, type: type, source: source, data: data, requestEvent: nil)
+    public convenience init(name: String, type: String, source: String, data: [String: Any]?, mask: [String]? = nil) {
```

Comment:
I think it might be a breaking change for Core ABIs. 
Should we add ad new convenience initializer instead of modifying this one?